---
title: Transcriptome-wide association study (TWAS)
author: JSH
date: 2024-04-20 20:55:00 +0800
categories: [Study, Bioinformatics]
tags: [Study, Bioinformatics]
---

## Transcriptome-wide association study (TWAS)
TWAS is a gene-prioritization approach that detects trait-associated genes regulated by significant variants, primarily single nucleotide polymorphism (SNP) identified from genome-wide association study (GWAS). 
Briefly, TWAS first trains a genetic regulation model of genetic components and gene expression from a small available reference panel. 
These models with regulatory weights are used to impute gene expression for individuals of larger GWAS cohorts. 
Finally, the associations between predictive gene expression and traits are calculated to determine the regulatory relationship between genes and traits. 

![image](https://github.com/JeonSHyun/JeonSHyun.github.io/assets/86886562/e6fce053-8854-4f87-888e-31abac96c106)

### Advantages of TWAS

#### Higher gene-based interpretability than GWAS
Although GWAS has identified numerous significant variants, most are in non-coding regions and are hard to interpret. As genes are functioning and more explainable units than variants, the gene-trait associations detected by TWAS are more sensible for explaining the genetic mechanism of phenotypes. 

#### Lower computing complexity and experimental cost
Compared to genome loci-based analysis, TWAS conducts association tests only for genes significantly regulated by genetic variations, thus relieving the multiple testing burden. Besides, as it narrows the range of candidate genes, this method can save time and labor costs for following experiments.

#### Tissue specificity
Most disease investigations are based on specific pathological tissues. By targeting relevant and functional tissues, TWAS detects trait-associated gene panels to better reveal the underlying pathological mechanisms for diverse diseases in a tissue-specific manner.

#### Higher statistical power than other gene-based analyses
Facing limited samples with expression data due to finite bio-samples and the high cost of RNA sequencing, TWAS highlights imputing gene expression of large-scale GWAS samples and thus significantly improves the statistical ability of association testing.

#### Leverage genetic regulation information
TWAS builds an expression imputation model from expression quantitative loci (eQTL) data and thus identifies genetically-regulated genes associated with traits, even if they are far from the variants.


### Workflow and updating models of TWAS

#### Training
This step aims to estimate regulatory effect sizes of multiple SNPs on the gene expression level by fitting a multivariate SNP ~gene expression model from a small reference panel with both genotype and expression data. 
Many eQTL databases and specific data sets are utilized to build the model.

#### Imputation
This step aims to obtain the predicted gene expression level of GWAS individuals. 
The trained prediction model takes large-scale genotype data from GWAS as inputs and outputs the calculated expression level of each gene. 
Imputation of expression might be combined into the association with GWAS summary statistics data. 
Generally, a reference panel of the same ancestry with GWAS samples is recommended to avoid disturbing unknown linkage disequilibrium (LD) profiles among diverse populations. 
Recently, multi-ancestry TWAS methods integrating eQTL data from multiple ethnicities have been developed, enabling cross-ethnic reference with a larger sample size.

#### Association
This step implements hypothesis tests between predicted gene expression and the target trait with different statistic association models. 
Finally, it pinpoints significant trait-associated genes with their effect sizes calculated. 
To avoid false positive results induced by multiple testing, statistical corrections are applied to adjust p-values, including Bonferroni correction, Benjamini-Hochberg correction, etc.


## Reference
[1] Mai, Jialin, et al. "Transcriptome-wide association studies: recent advances in methods, applications and available databases." Communications Biology 6.1 (2023): 899.
