---
title: Comparing variants related to chronic diseases from genome-wide association study (GWAS) and the cancer genome atlas (TCGA)
author: JSH
date: 2023-12-01 00:00:00 +0800
categories: [Publication, Article]
tags: [Article]
pin: true
---

![image](https://github.com/JeonSHyun/JeonSHyun.github.io/assets/86886562/e00d527d-2022-4b12-a70f-d0d32320de2f)

## Authors
**Soohyun Jeon***, Chaewon Park*, Jineui Kim, Jung Hoon Lee, Sung-yune Joe, Young Kyung Ko, Jeong-An Gim

## Abstract
### Background
Several genome-wide association studies (GWAS) have been performed to identify variants related to chronic diseases. Somatic variants in cancer tissues are associated with cancer development and prognosis. Expression quantitative trait loci (eQTL) and methylation QTL (mQTL) analyses were performed on chronic disease-related variants in TCGA dataset.
### Methods
MuTect2 calling variants for 33 carcinomas from TCGA and 296 GWAS variants provided by LocusZoom were used. At least one mutation was found in TCGA 22 carcinomas and LocusZoom 23 studies. Differentially expressed genes (DEGs) and differentially methylated regions (DMRs) from the three carcinomas (TCGA-COAD, TCGA-STAD, and TCGA-UCEC). Variants were mapped to the world map using population locations of the 1000 Genomes Project (1GP) populations. Decision tree analysis was performed on the discovered features and survival analysis was performed according to the cluster.
### Results
Based on the DEGs and DMRs with clinical data, the decision tree model classified seven and three nodes in TCGA-COAD and TCGA-STAD, respectively. A total of 11 variants were commonly detected from TCGA and LocusZoom, and eight variants were selected from the 1GP variants, and the distribution patterns were visualized on the world map.
### Conclusions
Variants related to tumors and chronic diseases were selected, and their geological regional 1GP-based proportions are presented. The variant distribution patterns could provide clues for regional clinical trial designs and personalized medicine.

## Status
Published

## Access
[Link](https://link.springer.com/article/10.1186/s12920-023-01758-7?utm_source=rct_congratemailt&utm_medium=email&utm_campaign=oa_20231219&utm_content=10.1186/s12920-023-01758-7)

## Preprint
[Link](https://www.researchsquare.com/article/rs-2492450)
