---
title: Genome-wide association study (GWAS)
author: JSH
date: 2024-04-20 20:55:00 +0800
categories: [Study, Bioinformatics]
tags: [Study, Bioinformatics]
---

## Genome-wide association study (GWAS)
GWAS test hundreds of thousands of genetic variants across many genomes to find those statistically associated with a specific trait or disease. 
This methodology has generated a myriad of robust associations for a range of traits and diseases, and the number of associated variants is expected to grow steadily as GWAS sample sizes increase. 
GWAS results have a range of applications, such as gaining insight into a phenotype’s underlying biology, estimating its heritability, calculating genetic correlations, making clinical risk predictions, informing drug development programmes and inferring potential causal relationships between risk factors and health outcomes.

![image](https://github.com/JeonSHyun/JeonSHyun.github.io/assets/86886562/da7cdbe3-a80c-42c7-abf2-451874431d7e)

## Conducting GWAS

### Selecting study populations
GWAS often require very large sample sizes to identify reproducible genome-wide significant associations. 
Study designs can involve the inclusion of cases and controls when the trait of interest is dichotomous, or quantitative measurements on the whole study sample when the trait is quantitative. 
In addition, one can choose between population-based and family-based designs. 
The choice of data resource and study design for a GWAS depends on the required sample size, the experimental question and the availability of pre-existing data or the ease with which new data can be collected. 
GWAS can be conducted using data from resources such as biobanks or cohorts with disease-focused or population-based recruitment, or through direct to consumer studies. 

For all study designs, recruitment strategies must be carefully considered as these can induce collider bias and other forms of bias in the resultant data.


### Genotyping
Genotyping of individuals is typically done using microarrays for common variants or next-generation sequencing methods such as WES or WGS that also include rare variants. 
Microarray-based genotyping is the most commonly used method for obtaining genotypes for GWAS owing to the current cost of next-generation sequencing.


### Data processing
Input files for a GWAS include anonymized individual ID numbers, coded family relations between individuals, sex, phenotype information, covariates, genotype calls for all called variants and information on the genotyping batch. 
Following input of the data, generating reliable results from GWAS requires careful quality control. 

Some example steps include removing rare or monomorphic variants, removing variants that are not in Hardy–Weinberg equilibrium, filtering SNPs that are missing from a fraction of individuals in the cohort, identifying and removing genotyping errors, and ensuring that phenotypes are well matched with genetic data, often by comparing self-reported sex versus sex based on the X and Y chromosomes. 
Software tools such as PLINK have been specifically designed to analyse genetic data and can be used to conduct many of these quality control steps.

Once sample and variant quality control have been performed on GWAS array data, variants usually undergo phasing and are imputed using a sequenced haplotype reference panel such as the 1000 Genomes Project or TOPMed, which involves the statistical inference of genotypes that have not been assayed directly. 
GWAS consortia routinely follow pipelines for conducting quality control steps and imputation, using, for example, RICOPILI23 or similar software, or upload their data to imputation servers (e.g., the Michigan Imputation Server or the TOPMed Imputation Server) where these standardized pipelines have been implemented. 
To achieve the large sample sizes typical in genetic studies in a logistically feasible manner that follows data protection rules, the above steps are often done separately for many different cohorts of varying sample size.

Ancestry and relatedness must be carefully considered and accounted for in GWAS, and indeed all genetic studies — particularly in data sets from participants of diverse backgrounds to avoid false positive or negative genetic signals and biased test statistics owing to population stratification.
Cases and controls should be matched by ancestry to avoid confounding; 
for example, a GWAS for chopstick use where cases are defined as ‘using chopsticks regularly’ and controls as ‘not using chopsticks’ would likely result in cases being drawn more often from an East Asian population than controls. 
Ancestry is usually considered in GWAS through an iterative process using principal component analysis.
This is done first to identify and exclude outliers, and then to compute and include principal components as covariates in subsequent GWAS regression models.


### Testing for associations
Typically in GWAS, linear or logistic regression models are used to test for associations, depending on whether the phenotype is continuous (e.g., height, blood pressure, body mass index) or binary (e.g., presence or absence of disease), respectively. 
Covariates such as age, sex and ancestry are included to account for stratification and avoid confounding effects from demographic factors. 
Including an additional random effect term — which is individual-specific in linear or logistic mixed models to account for genetic relatedness among individuals — can improve statistical power for genomic discovery and increase control for stratification.
When conducting a GWAS, it should be noted that the genotypes of genetic variants that are physically close together are not independent as they tend to be in linkage disequilibrium; 
this dependency of tests should also be considered when conducting a GWAS.

Linear regression models for GWAS:
![image](https://github.com/JeonSHyun/JeonSHyun.github.io/assets/86886562/f1a52180-95eb-42ec-98ba-f02066dc5c9f)


### Accounting for false discovery
Testing millions of associations between individual genetic variants and a phenotype of interest requires a stringent multiple-testing threshold to avoid false positives. 
The appropriate threshold might vary depending on the population.
Complex traits tend to be highly polygenic and, as a result, many genetic variants with small effects contribute to the phenotype. 
In these cases, winner’s curse is common, and effect size estimates close to the discovery threshold tend to be overestimated in initial GWAS.

Comparing effect sizes between discovery and independent replication cohorts is the gold standard for accounting for false discovery and winner’s curse by calibrating effect size estimates. 
Replication cohorts are ideally considered at the outset of the GWAS and should give sufficient statistical power to correct for winner’s curse and multiple testing. 
Replication cohorts must be completely independent from the discovery cohort, with no shared individuals or genetic relationships between individuals from the cohorts.


## Reference
[1] https://www.nature.com/articles/s43586-021-00056-9
